components/Pages/DocuSignPage/CartTotal.tsx

```diff
@@ -13,14 +14,27 @@ const CartTotal = () => {
   const { setIsDocuSignModalOpen } = useModalProvider()
   const { baseUrl } = useBaseUrl()
   const router = useRouter()
-
+  const [contractId, setContractId] = useState<string>('')
   const handleCheckout = () => {
     handleStripeCheckout({
       baseUrl,
       amount: (pricePerContract + taxPerContract) * 100,
+      metadata: {
+        cid: contractId,
+      },
     })
   }
 
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search)
+    const contractIdParamValue = searchParams.get('contractId') || ''
+    if (contractIdParamValue) {
+      setContractId(contractIdParamValue)
+    } else {
+      router.push('/')
+    }
+  }, [router])
```

Comment:
Why not use `useParams` from NextJS instead of this manual useEffect?

![Screenshot 2024-11-05 at 9 15 40 PM](https://github.com/user-attachments/assets/37c10029-4a77-4888-ae91-57bc54f08193)

https://chatgpt.com/share/672ab534-67cc-8009-9aa1-8ef71f7f55d6